src/main/scala/com/gu/thrift/serializer/Compression.scala

```diff
@@ -8,37 +10,51 @@ import scala.annotation.tailrec
 sealed trait CompressionType
 case object NoneType extends CompressionType
 case object GzipType extends CompressionType
+case object ZstdType extends CompressionType
 
-object Compression {
-
-  def gzip(data: Array[Byte]): Array[Byte] = {
-    try {
-        val bos = new ByteArrayOutputStream()
-        val out = new GZIPOutputStream(bos)
-        out.write(data)
-        out.close()
-        bos.toByteArray
-    } catch {
-      case e: IOException => throw new RuntimeException(e);
-    }
+object GzipCompression {
+
+  def compress(data: Array[Byte]): Array[Byte] = {
+    val bos = new ByteArrayOutputStream()
+    val out = new GZIPOutputStream(bos)
+    out.write(data)
+    out.close()
+    bos.toByteArray
   }
 
-  def gunzip(data: Array[Byte]): Array[Byte] = {
-    try {
-      val bos = new ByteArrayOutputStream()
-      val bis = new ByteArrayInputStream(data)
-      val in = new GZIPInputStream(bis)
-      copy(in, bos)
-      in.close()
-      bos.close()
-      bos.toByteArray()
-    } catch {
-      case e: IOException => throw new RuntimeException(e);
-    }
+  def uncompress(data: Array[Byte]): Array[Byte] = {
+    val bos = new ByteArrayOutputStream()
+    val bis = new ByteArrayInputStream(data)
+    val in = new GZIPInputStream(bis)
+    IOUtils.copy(in, bos)
+    in.close()
+    bos.close()
+    bos.toByteArray()
   }
+}
 
+object ZstdCompression {
+  def compress(data: Array[Byte]): Array[Byte] = {
+    val bos = new ByteArrayOutputStream()
+    val out = new ZstdOutputStream(bos)
+    out.write(data)
+    out.close()
+    bos.toByteArray
```

Comment:
I feel like there should be some `try { ... } finally { ... }` stuff in here to ensure that we don't leak resources. However it's likely to be in memory, so perhaps it's not possible to leak resources here.